File: c++/Array/DupWithinK.cpp
#include<iostream>
#include<vector>
#include<unordered_set>
using namespace std;
int DupWithinK(vector<int> arr,int k)
{
    int res=0;
    unordered_set<int> mp;
    for(int i=0;i<arr.size();i++)
    {
        if(mp.find(arr[i])!=mp.end())
        {
            res++;
            cout<<i<<endl;
        }
            mp.insert(arr[i]);
            if(i>=k)
                mp.erase(arr[i-k]);
    }
    //for(int i=0;i<mp.size();i++)
    //    cout<<i<<" "<<mp[i]<<endl;
    return res;
}
int main(){
    int k=3;
    vector<int> arr={1,2,3,1,4,5,2,4};
    cout<<DupWithinK(arr,k);
return 0;
}

File: java/Array/DupWithInK.java
import java.util.*;
class DupWithInK
{
    public static int DupInK(int[] arr,int k)
    {
        HashSet<Integer> Set=new HashSet<Integer>();
        int res=0;
        for(int i=0;i<arr.length;i++)
        {
            if(Set.contains(arr[i]))
            {
                res++;
            }
            Set.add(arr[i]);
            if(i>=k)
            Set.remove(arr[i-k]);
        }
        return res;
    }
    public static void main(String[] args) {
        int n;
        Scanner sc=new Scanner(System.in);
        n=sc.nextInt();
        int[] arr=new int[n];
        for(int i=0;i<n;i++)
        {
            arr[i]=sc.nextInt();
        }
        int k=sc.nextInt();
        System.out.println(DupInK(arr, k));
    }
}